### PR TITLE
Remove uses of JS classes in OCaml, part 2

### PR DESCRIPTION
**Pull request by mitschabaude**
bindings: https://github.com/o1-labs/snarkyjs-bindings/pull/36

follow up to #956 

* Removes the old `Field` and `Bool`
* Moves OCaml side of Poseidon into the `Snarky` namespace, so that we only have 4 namespaces to manage - `Snarky`, `Pickles`, `Ledger`, `Test`
* Move some of the `Ledger` static methods into `Test`
* Removed some unused stuff

### CODE REVIEW

File: src/lib/bool.ts [hunk 1]
```diff
@@ -311,37 +310,31 @@ class Bool {
     return 1;
   }
 
-  // TODO
-  static count(x: Bool | boolean[]): Field {
```
Comment (mitschabaude) [inline src/lib/bool.ts:315]: removed unimplemented function

File: src/lib/bool.ts [hunk 2]
```diff
@@ -311,37 +310,31 @@ class Bool {
...
-      if (typeof x === 'number') {
-        return new Bool(x === 1);
-      } else if (typeof x === 'string') {
-        return new Bool(x === '1');
-      } else if (typeof x === 'boolean') {
-        return new Bool(x);
-      } else {
-        return new Bool(x.value);
-      }
+    ofField(x: Field) {
```
Comment (mitschabaude) [inline src/lib/bool.ts:326]: I refactored this function as I felt the additional options to pass in non-`Field`s were never used and made it less safe, and I added a runtime error

**Review by mitschabaude — COMMENTED** [review]
some comments!

**Review by MartinMinkov — APPROVED** [review]
Looks great! :partying_face:

**Pull request merged**